certbot/certbot/_internal/main.py

```diff
@@ -1652,6 +1653,76 @@ def make_or_verify_needed_dirs(config: configuration.NamespaceConfig) -> None:
         util.make_or_verify_dir(hook_dir, strict=config.strict_permissions)
 
 
+def reconfigure(config: configuration.NamespaceConfig,
+          plugins: plugins_disco.PluginsRegistry) -> None:
+    """Allow the user to set new configuration options for an existing certificate without
+       forcing renewal. This can be used for things like authenticator, installer, and hooks,
+       but not for the domains on the cert, since those are only saved in the cert.
+
+    :param config: Configuration object
+    :type config: configuration.NamespaceConfig
+
+    :param plugins: List of plugins
+    :type plugins: plugins_disco.PluginsRegistry
+
+    :raises errors.Error: if the dry run fails
+    :raises errors.ConfigurationError: if certificate could not be loaded
+
+    """
+
```

Comment:
I think "changes apply at the next renewal" is the way to go. I assume the distinction you were thinking was "these will apply to the cert itself" vs. "these apply to the webserver/local configuration"? Even if it applies to the cert itself, I can see it being useful to decouple changing what you want it to do and actually doing it in cases where rate limits might apply or some such.

Reply:
Would a renewal, ahead of usual renewal time window, with updated `--key-type` et cetera require a `--force-renewal`? Or does Certbot notice differences between the current cert and `reconfigure`'d renewal param conf file, so that `--force-renewal` isn't necessary? I think it would be nice if Certbot noticed.

Reply:
It currently just checks ocsp (maybe) and cert expiry date, `--force-renewal` is necessary.

I currently added `Changes will apply when the certificate renews.` if anything was changed. Debating mentioning `--force-renewal` here, but leanings towards not doing so.

I agree noticing those differences would be nice to have, but I think it's not the simplest thing to do unless again we start listing particular terms we want here.